tools/plugin-toolkit/src/_base/credentials.ts

```diff
@@ -0,0 +1,70 @@
+// (C) 2021 GoodData Corporation
+
+import { InputValidationError, TargetBackendType } from "./types";
+import isEmpty from "lodash/isEmpty";
+
+export type BackendCredentials = {
+    /**
+     * Value of GDC_USERNAME retrieved from the environment (files or process)
+     */
+    username: string | undefined;
+
+    /**
+     * Value of GDC_PASSWORD retrieved from the environment (files or process)
+     */
+    password: string | undefined;
+
+    /**
+     * Value of TIGER_API_TOKEN retrieved from the environment (files or process)
+     */
+    token: string | undefined;
+};
+
+export function createCredentialsFromEnv(env: Record<string, string>): BackendCredentials {
+    return {
+        username: env.GDC_USERNAME,
+        password: env.GDC_PASSWORD,
+        token: env.TIGER_API_TOKEN,
+    };
+}
+
+/**
+ * Validate that the gathered credentials provided on the input are complete in the context of the
+ * specified backend type. For 'bear' both username & password must be specified. For 'tiger' only
+ * token must be specified.
+ *
+ * If any essential prop is missing value - being undefined or empty, then InputValidationError flies. Otherwise
+ * the function just returns.
+ *
+ * @param backend - backend being targeted by the CLI
+ * @param credentials - gathered credentials
+ */
+export function validateCredentialsAreComplete(
+    backend: TargetBackendType,
+    credentials: BackendCredentials,
+): void {
+    const { username, password, token } = credentials;
+
+    if (backend === "bear") {
+        if (isEmpty(username)) {
+            throw new InputValidationError(
+                "username",
+                "",
+                "Unable to determine username to use when logging into GoodData platform. Please make sure GDC_USERNAME env variable is set in your session or in the .env file",
```

Comment:
These messages should mention .env.secrets file right?

Reply:
fixed